pydm/widgets/base.py

```diff
@@ -334,6 +333,39 @@ def enum_strings_changed(self, new_enum_strings):
             self.enum_strings = new_enum_strings
             self.value_changed(self.value)
 
+    def eventFilter(self, obj, event):
+        """
+        EventFilter to redirect "middle click" to :meth:`.show_address_tooltip`
+        """
+        # Override the eventFilter to capture all middle mouse button events,
+        # and show a tooltip if needed.
+        if event.type() == QEvent.MouseButtonPress:
+            if event.button() == Qt.MiddleButton:
+                self.show_address_tooltip(obj, event)
+                return True
+        return False
+
+    def show_address_tooltip(self, event):
+        """
+        Show the PyDMTooltip and copy address to clipboard
+
+        This is intended to replicate the behavior of the "middle click" from
+        EDM. If the QWidget does not have a valid PyDMChannel nothing will be
+        displayed
+        """
+        if not len(self._channels):
+            logger.warning("Object %r has no PyDM Channels", obj)
+            return
+        addr = self.channels()[0].address
+        QToolTip.showText(event.globalPos(), addr)
+        # If the address has a protocol, and it is the default protocol, strip
+        # it out before putting it on the clipboard.
+        m = re.match('(.+?):/{2,3}(.+?)$', addr)
```

Comment:
Seeing these sorts of lines repeated makes me think that some parsing functions in utils would be useful.

Reply:
Addressed on the last commit 👍